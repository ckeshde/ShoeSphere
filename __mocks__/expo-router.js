// __mocks__/expo-router.js
export const useRouter = () => ({
  push: jest.fn(),
});
export const useLocalSearchParams = jest.fn(() => ({}));
