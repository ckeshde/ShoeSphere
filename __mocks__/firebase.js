export const collection = jest.fn();
export const getDocs = jest.fn();
export const addDoc = jest.fn();
export const deleteDoc = jest.fn();
export const doc = jest.fn();
export const onSnapshot = jest.fn();
export const query = jest.fn();
export const where = jest.fn();

export const db = {};
