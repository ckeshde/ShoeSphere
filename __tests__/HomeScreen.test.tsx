import React from 'react';
import renderer from 'react-test-renderer'; // ðŸ‘ˆ Add this for snapshot testing
import { render, waitFor } from '@testing-library/react-native';
import HomeScreen from '../app/(tabs)/index';
import { onAuthStateChanged, getAuth } from 'firebase/auth';
import { useRouter } from 'expo-router';

// Mock Firebase Auth
jest.mock('firebase/auth', () => ({
  getAuth: jest.fn(),
  onAuthStateChanged: jest.fn(),
}));

// Mock navigation
jest.mock('expo-router', () => ({
  useRouter: jest.fn(),
}));

// Mock StoreSearch to avoid testing it here
jest.mock('../app/pages/storesearch', () => {
  const React = require('react');
  const { Text } = require('react-native');
  return () => <Text>Mock StoreSearch</Text>;
});

describe('HomeScreen', () => {
  const mockReplace = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (useRouter as jest.Mock).mockReturnValue({ replace: mockReplace });
  });

  it('redirects to /login if user is not logged in', async () => {
    const fakeUnsubscribe = jest.fn();
    (onAuthStateChanged as jest.Mock).mockImplementation((auth, callback) => {
      callback(null); // simulate no user
      return fakeUnsubscribe;
    });

    render(<HomeScreen />);

    await waitFor(() => {
      expect(mockReplace).toHaveBeenCalledWith('/login');
    });
  });

  it('redirects to /tabs if user is logged in', async () => {
    const fakeUnsubscribe = jest.fn();
    const mockUser = { uid: '123' };
    (onAuthStateChanged as jest.Mock).mockImplementation((auth, callback) => {
      callback(mockUser); // simulate user
      return fakeUnsubscribe;
    });

    render(<HomeScreen />);

    await waitFor(() => {
      expect(mockReplace).toHaveBeenCalledWith('/(tabs)');
    });
  });

  it('renders StoreSearch component', async () => {
    const fakeUnsubscribe = jest.fn();
    const mockUser = { uid: '123' };
    (onAuthStateChanged as jest.Mock).mockImplementation((auth, callback) => {
      callback(mockUser);
      return fakeUnsubscribe;
    });

    const { getByText } = render(<HomeScreen />);
    expect(getByText('Mock StoreSearch')).toBeTruthy();
  });

  // âœ… Snapshot Test
  it('matches the snapshot', () => {
    const mockUser = { uid: '123' };
    (onAuthStateChanged as jest.Mock).mockImplementation((auth, callback) => {
      callback(mockUser);
      return jest.fn();
    });

    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
