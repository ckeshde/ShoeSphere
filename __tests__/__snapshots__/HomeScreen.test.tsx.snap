// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
    }
  }
>
  <Text>
    Mock StoreSearch
  </Text>
</View>
`;
